src/main/java/org/opensearch/knn/index/query/KNNWeight.java

```diff
@@ -185,13 +169,21 @@ private BitSet getFilteredDocsBitSet(final LeafReaderContext ctx) throws IOExcep
         return createBitSet(scorer.iterator(), liveDocs, maxDoc);
     }
 
-    private BitSet createBitSet(final DocIdSetIterator filteredDocIdsIterator, final Bits liveDocs, int maxDoc) throws IOException {
-        if (liveDocs == null && filteredDocIdsIterator instanceof BitSetIterator) {
+    private BitSet getAllDocsBitSet(final LeafReaderContext ctx) throws IOException {
+        final FloatVectorValues floatVectorValues = ctx.reader().getFloatVectorValues(this.knnQuery.getField());
```

Comment:
There can be ByteVectorValues too. You should actually create KNNVectorValues here using KNNVectorValuesFactory class.